### PR TITLE
Refactor packager.ts into smaller functions

### PR DESCRIPTION
**Pull request by mikelovesrobots**
Hi!

I'm writing a book on refactoring. To test out the methods I promote in my book, I’ve been working on open-source projects I admire, identifying complex files and simplifying them.

In this repo, I ran `git log --name-only --pretty=format: | sort | uniq -c | sort -rg | head -40` and found that packager.ts was the most frequently modified file, with around 50 commits. It’s also over 100 lines long, so I refactored it by breaking it into smaller, single-purpose functions. Each function is unit-tested, and there are no actual functional changes.

The existing code is already pretty strong, but this refactor on that one hotspot should make it easier to work with in the future. This PR is unsolicited and my first time working with your coding conventions, so let me know if anything needs adjustment.

Thanks for considering this!

## Functional changes

None. This PR only refactors packager.ts by extracting out smaller functions.

## Checklist

- [x] Run `npm run test`
- [x] Run `npm run lint`
  - Note: Some files outside the scope of this PR fail the linter. Just wanted to flag that.

### CODE REVIEW

**Comment by yamadashy** [comment]
Hi, @mikelovesrobots !
Thank you for this refactoring!

You're right - packager.ts was frequently modified and caused conflicts across multiple PRs.

I fixed a small linting error but your code changes look perfect. While the metrics-related splitting might seem a bit granular, I think it's actually well-suited for AI-assisted code generation, so I'm satisfied with this approach.
Thanks for adding the comprehensive tests too.

I'll merge this in!

File: tests/core/metrics/aggregateMetrics.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { describe, expect, it } from 'vitest';
+import type { FileMetrics } from '../../../lib/core/metrics/calculateIndividualFileMetrics.js';
```
Comment (huy-trn) [inline tests/core/metrics/aggregateMetrics.test.ts:2]: This line should be:
```typescript
import type { FileMetrics } from '../../../src/core/metrics/calculateIndividualFileMetrics.js';
```

Comment (huy-trn) [reply tests/core/metrics/aggregateMetrics.test.ts:2]: Hey guys, I saw CI was failing so took a quick look and found this. Hope it helps!

Comment (mikelovesrobots) [reply tests/core/metrics/aggregateMetrics.test.ts:2]: Thanks for catching that. Not sure how it was passing here, but yeah, certainly breaking main.

Here's the fix:
https://github.com/yamadashy/repomix/pull/218

**Comment by yamadashy** [comment]
@mikelovesrobots 
Released in v0.2.7! 🎉

https://github.com/yamadashy/repomix/releases/tag/v0.2.7

While this change is classified as "internal", it greatly improves maintainability for future development.
Thank you for your excellent contribution! Please feel free to contribute again anytime:

**Pull request merged**